src/state/trajectory/logics.ts

```diff
@@ -637,6 +661,34 @@ const cancelConversionLogic = createLogic({
     type: CANCEL_CONVERSION,
 });
 
+const receiveDefaultUIDataLogic = createLogic({
+    process(deps: ReduxLogicDeps, dispatch, done) {
+        const { getState, action } = deps;
+        const browserStoredUIData = getUserSelectedUIData(getState());
+        /**
+         * If the conditions below are true, then valid color settings were
+         * retrieved from browser storage before the default ui data arrived
+         * and those settings should be applied.
+         * If false then userSelected data either doesn't exist, hasn't arrived yet, or
+         * is invalid: set to empty array for now.
+         */
+        if (
+            browserStoredUIData.length > 0 &&
+            compareAgentTrees(browserStoredUIData, action.payload)
+        ) {
+            dispatch(
+                setCurrentColorSettings({
+                    currentColorSettings: ColorSettings.UserSelected,
+                })
+            );
+        } else {
+            dispatch(setUserSelectedUIData([]));
+        }
+        done();
+    },
+    type: SET_DEFAULT_UI_DATA,
+});
+
```

Comment:
One of two places we are handling checks for different trajectories with the same name.

If browser settings were retrieved from storage before this logic ran, we did not apply them as we couldn't verify if they were appropriate for the current trajectory in a rigorous way. We do the check here and apply them if so. If the check fails, we restore initial state for userSelectedUIData to make way for incoming color changes or settings.